caikit/runtime/http_server/request_aborter.py

```diff
@@ -0,0 +1,108 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This module helps us know when a HTTP client call is cancelled, and we need to stop or undo work
+"""
+# Standard
+from typing import Optional
+import asyncio
+import threading
+
+# Third Party
+from fastapi import Request
+
+# First Party
+import alog
+
+# Local
+from caikit.runtime.work_management.abortable_action import ActionAborter
+
+log = alog.use_channel("REQUEST-ABORTER")
+
+
+class HttpRequestAborter(ActionAborter):
+    """
+    In order to actually interrupt threads doing the work, events can be registered with an
+    instance of this class in order ton receive notification on request disconnection. This allows
+    work to be terminated when a client time's out or stops listening.
+
+    IFF the client request has been terminated, `must_abort` will return True.
+    """
+
+    def __init__(
+        self,
+        context: Request,
+        loop: Optional[asyncio.AbstractEventLoop] = None,
+        poll_time: Optional[float] = 1,
```

Comment:
It seems like we might want to make this default a little smaller since many requests will not last much longer than 1s even if not aborted.

Reply:
Can do! I'm coming from a land where the average response time can be as high as 75s...